pkg/base/helm.go

```diff
@@ -271,3 +274,105 @@ func checkChartForVersion(file *upstreamtypes.UpstreamFile) (string, error) {
 	// if no determination is made, assume v2
 	return "v2", nil
 }
+
+// insert namespace if it's defined in the spec and not already present in the manifests
+func kustomizeHelmNamespace(baseFiles []BaseFile, renderOptions *RenderOptions) ([]BaseFile, error) {
```

Comment:
this got very complicated. since there are potentially duplicates (i.e. patches), i still had to filter those and add them to `kustomization.yaml` correctly. that means those patches are already applied so i chose to not pass them on to the next step. for resources that have a hardcoded namespace, duplicates are maintained and added to base like before.